decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ] I this you should add a setter method to your nameable attribute.  This way,  the attribute will be accessible and modifiable  outside the Decorator class

e.g.

`attr_accessor :nameable`